functions/configfile.pp

```diff
@@ -0,0 +1,10 @@
+# Generates configuration files for mcollective. Keys are sorted alphabetically:
+# # Configuration is managaged by puppet class main
+# key = valze
+function mcollective::configfile(Hash $confighash) >> String {
+  $sorted = $confighash.keys.sort
+  $result = $sorted.map |$key| {
+    "${key} = ${confighash[$key]}"
+  }
+  (["# Configuration is managed by puppet class ${name}"] << $result << []).join("\n")
```

Comment:
This ends up beieng resolved only as `main`. tried using my usual `$module_name` as prefix, but this does not resolve it seems. Should we:

- Rather add prefix, `mcollective::`
- Have only one static string
- Remove any hint
- or even make it a function parameter?